### PR TITLE
Prevent NullPointerException when data is null during a Payment

### PR DESCRIPTION
**Pull request by xsellier**
Fixes https://github.com/godotengine/godot/issues/17481

### CODE REVIEW

File: platform/android/java/src/org/godotengine/godot/payments/HandlePurchaseTask.java [hunk 1]
```diff
@@ -56,6 +56,12 @@ public HandlePurchaseTask(Activity context) {
 	}
 
 	public void handlePurchaseRequest(int resultCode, Intent data) {
+
+		if (data == null) {
+			canceled();
+			return;
+    }
```
[the remark this answers is not in the document]
Comment (xsellier) [reply platform/android/java/src/org/godotengine/godot/payments/HandlePurchaseTask.java]: Done

**Comment by volzhs** [comment]
it would be great to check the `resultCode` in #17481 case
i'd like to try, but don't have time to test it.

**Comment by xsellier** [comment]
@volzhs In fact this fix is way better. We need to check the `resultCode` before trying to access to the intent.

**Comment by volzhs** [comment]
@xsellier thanks for the updating.
@megasoft78 would you check this PR solve your case?

**Comment by megasoft78** [comment]
@volzhs @xsellier I'll test it together with the other fixes on the next release. :)

**Comment by volzhs** [comment]
@megasoft78 cool. :)

**Pull request merged**